### PR TITLE
[DOC] Enhanced RDoc for MatchData

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #to_s
- #named_captures
- #string
- #inspect
- #hash
- #==

Also some very minor housekeeping.

### CODE REVIEW

File: re.c [hunk 1]
```diff
@@ -2337,12 +2335,14 @@ match_named_captures(VALUE match)
 
 /*
  *  call-seq:
- *     mtch.string   -> str
+ *    string -> string
  *
- *  Returns a frozen copy of the string passed in to <code>match</code>.
+ *  Returns a frozen copy of the target string:
```
Comment (jeremyevans) [inline re.c:2340]: This isn't accurate if the string passed in is already frozen (the existing documentation was wrong).  In that case, it returns the string passed in, not a frozen copy.

File: re.c [hunk 2]
```diff
@@ -2373,21 +2373,26 @@ match_inspect_name_iter(const OnigUChar *name, const OnigUChar *name_end,
...
  *
- * Returns a printable version of <i>mtch</i>.
+ *    m /.$/.match("foo")
+ *    m.inspect # => "#<MatchData \"o\">"
  *
- *     puts /.$/.match("foo").inspect
- *     #=> #<MatchData "o">
+ *    /(.)(.)(.)/.match("foo")
+ *    # => #<MatchData "foo" 1:"f" 2:"o" 3:"o">
+ *    m.inspect # => "#<MatchData \"o\">"
```
Comment (jeremyevans) [inline re.c:2385]: This line and other `m.inspect` lines below look wrong.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good, a few minor changes requested.

File: re.c [hunk 3]
```diff
@@ -2296,25 +2304,15 @@ match_named_captures_iter(const OnigUChar *name, const OnigUChar *name_end,
...
- *     mtch.named_captures -> hash
- *
- *  Returns a Hash using named capture.
+ *    named_captures -> hash
  *
- *  A key of the hash is a name of the named captures.
- *  A value of the hash is a string of last successful capture of corresponding
- *  group.
+ *  Returns a hash of the named captures;
+ *  each key is a capture name; each value is the captured string:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply re.c]: Fixed.

File: re.c [hunk 4]
```diff
@@ -2296,25 +2304,15 @@ match_named_captures_iter(const OnigUChar *name, const OnigUChar *name_end,
...
  *
- *     m = /(?<a>.)(?<b>.)/.match("01")
- *     m.named_captures #=> {"a" => "0", "b" => "1"}
- *
- *     m = /(?<a>.)(?<b>.)?/.match("0")
- *     m.named_captures #=> {"a" => "0", "b" => nil}
- *
- *     m = /(?<a>.)(?<a>.)/.match("01")
- *     m.named_captures #=> {"a" => "1"}
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply re.c:2314]: Fixed.

File: re.c (hunk 1 above)
Comment (BurdetteLamar) [reply re.c]: Fixed.

File: re.c (hunk 2 above)
Comment (BurdetteLamar) [reply re.c]: Fixed.

File: re.c [hunk 5]
```diff
@@ -2373,21 +2386,27 @@ match_inspect_name_iter(const OnigUChar *name, const OnigUChar *name_end,
...
  *
- *     puts /(.)(.)?(.)/.match("fo").inspect
- *     #=> #<MatchData "fo" 1:"f" 2:nil 3:"o">
+ *    m = /(.)(.)?(.)/.match("fo")
+ *    # => #<MatchData "fo" 1:"f" 2:nil 3:"o">
+ *    m.inspect # => "#<MatchData \"fo\" 1:\"f\" 2:nil 3:\"o\">"
  *
- *     puts /(?<foo>.)(?<bar>.)(?<baz>.)/.match("hoge").inspect
- *     #=> #<MatchData "hog" foo:"h" bar:"o" baz:"g">
+ *    m = /(?<foo>.)(?<bar>.)(?<baz>.)/
```
Comment (jeremyevans) [inline re.c:2405]: This example block shows `Regexp#inspect`, not `MatchData#inspect`, so I think it should be removed.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good.  There is one example block that should be deleted, after that this can be merged without additional review.

File: re.c (hunk 5 above)
Comment (BurdetteLamar) [reply re.c]: Done.

**Pull request merged**